src/rust/engine/src/externs/interface.rs

```diff
@@ -804,14 +806,19 @@ async fn workunit_to_py_value(
     externs::store_dict(py, artifact_entries)?,
   ));
 
-  if !workunit.counters.is_empty() {
-    let counters_entries = workunit
-      .counters
-      .iter()
+  // TODO: Temporarily attaching the global counters to the "root" workunit. Callers should
+  // switch to consuming `StreamingWorkunitContext.get_metrics`.
+  // Remove this deprecation after 2.14.0.dev0.
+  if workunit.parent_id.is_none() {
+    let mut metrics = workunit_store.get_metrics();
+
+    metrics.insert("DEPRECATED_ConsumeGlobalCountersInstead", 0);
```

Comment:
cc @asherf

Reply:
our backend don't expect any specific counters. but if counters name change we might need to update an internal metrics dashboard created by @Eric-Arellano

Reply:
looking at buildsense it seems that we no longer get metrics with this change.
so I think the proper way to make this change is to tweak our backend to work with the new counters (might need a new version of the tc pants plugin)  and only then merge this.
otherwise, we will loose pants metrics which we do rely on for some stuff.